cocos/base/utils/src/js.ts

```diff
@@ -68,6 +68,41 @@ export { IDGenerator } from './id-generator';
 export { Pool } from './pool';
 export { array };
 
+/**
+ * @en Inserts a new element into a map. All values corresponding to the same key are stored in an array.
+ * @zh 往 map 插入一个元素。同一个关键字对应的所有值存储在一个数组里。
+ * @param map @en The map to insert element. @zh 插入元素的 map。
+ * @param key @en The key of new element. @zh 新插入元素的关键字。
+ * @param value @en The value of new element. @zh 新插入元素的值。
+ * @param pushFront @en Whether to put new value in front of the vector if key exists. @zh 如果关键字已经存在，是否把新插入的值放到数组第一个位置。
+ *
+ * @example
+ * ```ts
+ * import { js, log } from 'cc';
+ *
+ * let a = { b: 1 };
+ * js.pushToRecord(a, 'b', 2, false);
+ * log(a);  // { b: [1, 2] }
+ * ```
+ */
+export function pushToRecord (map: Record<string, unknown>, key: string, value: unknown, pushFront: boolean): void {
```

Comment:
It seems map is more known to developers than record, and the parameter name is map too.

Reply:
updated https://github.com/cocos/cocos-engine/pull/16365/commits/bd7c2ad7e81bf379836c875ada6c3f156d27062b